### PR TITLE
Extended xcube.core.store.DataStore docstring to include a basic conv…

### LINKED ISSUE #330: Establish common data store conventions
**Is your feature request related to a problem? Please describe.**

The xcube data store framework should establish common conventions so we have a consistent behaviour across all store implementations. The current store implementations (SH, CCI, CDS) behave slightly different.

1. _Open parameters_ that constrain the dataset to be opened should all have the same names, have same or compatible type, and should be interpreted in the same way.
2. Stores should share the same idea of what is a _coordinate variable_ and what a _data variable_. 

**Describe the solution you'd like**

### 1. Open parameters

For a constraining open parameter P we introduce the following convention:

  * If _P is None_ means, parameter not given, hence no constraint applies, hence full containment.
  * If _not P_ means, we exclude what would otherwise be fully included.
  * Else, the given constraint applies.

Common names are `variable_names: List[str]`, `bbox:Union[str,Tuple[float, float, float, float]]`, `crs:str`, `spatial_res: float`, `time_range: Tuple[Optional[str], Optional[str]]`, `time_period: str`. A store may require all, or individual parameters, or none at all.

E.g. applied to `variable_names`, this means

  * `variable_names is None` - include all data variables
  * `variable_names == []` - do not include data variables (schema only)
  * `variable_names == ["<var_1>", "<var_2>", ...]` only include data variables named "<var_1>", "<var_2>", ...

Describe this convention in docstrings.

**EDIT**

A store implementation should document its available open parameters 

1. programmatically in the `DataStore.get_open_data_params_schema()`method
2. in the docstring of the implementing class. Each parameter description should include its name, Python type and description.

### 2. Coordinate vs Data Variables

* The open parameter `variable_names` shall refer to _data_ variables only.
* A store implementation shall add any available _coordinate_ variables at least for every dimension of the included data variables.

Describe this convention in docstrings.

**Additional context**

We'll create linked issues and PRs in the store code repos where we fix implementations wrt this issue.

**EDIT**

This issue resolves only with the following:

- [x] https://github.com/dcs4cop/xcube-cci/issues/10
- [x] https://github.com/dcs4cop/xcube-cds/issues/1
- [x] https://github.com/dcs4cop/xcube-sh/issues/12

### PR DESCRIPTION
**Pull request by forman**
…ention for store open parameters.

Closes #330

### CODE REVIEW

**Review by TonioF — COMMENTED** [review]
Data contraining open parameters would then be 'variable_names', 'bbox' and 'time_range' (other store-specific parameters may exist). As we have declared that these may be None, I think they should be set to optional in the list.

**Comment by forman** [comment]
> Data contraining open parameters would then be 'variable_names', 'bbox' and 'time_range' (other store-specific parameters may exist). As we have declared that these may be None, I think they should be set to optional in the list.

No, not necessarily. This is store dependent. Sentinel HUB for example does not define a useful default for `bbox` (whole world in 10m?) or `spatial_res` when we use a mixed set of Sentinel-2 bands.

**Review by TonioF — APPROVED** [review]
I approve, then.

**Review by AliceBalfanz — APPROVED** [review]

File: xcube/core/store/store.py [hunk 1]
```diff
@@ -228,6 +228,34 @@ def get_open_data_params_schema(self, data_id: str = None, opener_id: str = None
...
+        Given here are names, types, and descriptions of common, constraining open parameters for gridded datasets.
+        Note, whether any of these is optional or mandatory depends on the individual data store. A store may also
+        define other open parameters or support only a subset of the following:
+
+          * ``variable_names: List[str]``: Included data variables.
+             Available coordinate variables will be auto-included for any dimension of the data variables.
+          * ``bbox: Tuple[float, float, float, float]``: Spatial coverage.
+          * ``crs: str``: Spatial CRS, e.g. "EPSG:4326" or OGC CRS URI.
+          * ``spatial_res: float``: Spatial resolution in coordinates of the spatial CRS.
+          * ``time_range: Tuple[Optional[str], Optional[str]]``: Time range interval using iso-date/times.
```
Comment (pont-us) [inline xcube/core/store/store.py:250]: We should be explicit about the semantics of None in the time range (which I assume should be interpreted as "from the earliest data in the dataset" and "up to the latest data in the dataset" for start and end respectively).

**Comment by forman** [comment]
> Data contraining open parameters would then be 'variable_names', 'bbox' and 'time_range' (other store-specific parameters may exist). As we have declared that these may be None, I think they should be set to optional in the list.

Right.

File: xcube/core/store/store.py (hunk 1 above)
Comment (forman) [reply xcube/core/store/store.py:250]: Done.

File: xcube/core/store/store.py [hunk 2]
```diff
@@ -228,6 +228,34 @@ def get_open_data_params_schema(self, data_id: str = None, opener_id: str = None
...
+          * _if not P_ means, we exclude what would otherwise be fully included.
+          * _else_, the given constraint applies.
+
+        Given here are names, types, and descriptions of common, constraining open parameters for gridded datasets.
+        Note, whether any of these is optional or mandatory depends on the individual data store. A store may also
+        define other open parameters or support only a subset of the following:
+
+          * ``variable_names: List[str]``: Included data variables.
+             Available coordinate variables will be auto-included for any dimension of the data variables.
+          * ``bbox: Tuple[float, float, float, float]``: Spatial coverage.
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/core/store/store.py:247]: Done.

File: xcube/core/store/store.py [hunk 3]
```diff
@@ -228,6 +228,34 @@ def get_open_data_params_schema(self, data_id: str = None, opener_id: str = None
...
+        For maximum compatibility of stores, it is strongly encouraged to apply the following conventions on
+        parameter names, types, and their interpretation.
+
+        Let P be the value of an optional, data constraining open parameter,
+        then it should be interpreted as follows:
+
+          * _if P is None_ means, parameter not given, hence no constraint applies, hence full containment.
+          * _if not P_ means, we exclude what would otherwise be fully included.
+          * _else_, the given constraint applies.
+
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/core/store/store.py:240]: Done.

> Is it really intended to apply so broadly?

Why not?

**Pull request merged**